### PR TITLE
feat: e2e test cases of whole address book and inspection form

### PR DESCRIPTION
**Pull request by rashmi0301**
## Change Description

Done e2e test cases of address book and inspection form
Created the test cases for user tab as well

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rashmi0301** [comment]
@kunalfleethawks please review it .
Thanks

File: cypress/integration/address_broker.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
+
+
+it('added broker with required optional fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/#');
```
Comment (kunalfleethawks) [inline cypress/integration/address_broker.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_broker.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    cy.get('#brokerForm > :nth-child(1) > :nth-child(3) > :nth-child(3) > .row > .col-lg-12 > .form-control').type('4332543533');
+    //   cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-success').click();
+    cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-default').click();
+    cy.get('.address-book-icon').click();
+    /* ==== End Cypress Studio ==== */
+});
+
+it('added broker with required  fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/#');
```
Comment (kunalfleethawks) [inline cypress/integration/address_broker.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_broker.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    cy.get('.result-suggestions > ul').click();
+    cy.get('#broker > .mr-1 > :nth-child(2) > .mr-3').click();
+    cy.wait(8000);
+    cy.get('#broker > .mr-1 > :nth-child(2) > [type="button"]').click();
+    cy.wait(8000);
+});
+
+it('user be able to edit', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/#');
```
Comment (kunalfleethawks) [inline cypress/integration/address_broker.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_broker.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    cy.get('#brokerForm > :nth-child(1) > :nth-child(1) > :nth-child(3) > .row > :nth-child(5) > .form-control').type('9876543234');
+    // cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-success').click();
+    cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-success').click();
+    cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-default').click();
+    cy.get('.address-book-icon').click();
+});
+
+it('validation error on required  fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/#');
```
Comment (kunalfleethawks) [inline cypress/integration/address_broker.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_broker.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    cy.get('#brokerForm > :nth-child(1) > :nth-child(1) > :nth-child(3) > .row > :nth-child(3) > .form-control').clear();
+    cy.get('div:nth-of-type(3) > .ng-star-inserted.text-danger > .ng-star-inserted').contains('Email is required');
+    // cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-success').click();
+    cy.get('#addBrokerModal > .modal-dialog > .modal-content > .modal-footer > .btn-default').click();
+    cy.get('.address-book-icon').click();
+});
+
+it.only('user be able to delete', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/#');
```
Comment (kunalfleethawks) [inline cypress/integration/address_broker.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_customer.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
+
+
+it('added customer with required optional fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/');
```
Comment (kunalfleethawks) [inline cypress/integration/address_customer.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_customer.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    cy.get('#addCustomerModal > .modal-dialog > .modal-content > .modal-footer > .btn-default').click();
+    cy.get('.address-book-icon').click();
+    cy.get('#customer-tab').click();
+    cy.wait(6000);
+});
+
+
+it('added customer with required  fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/');
```
Comment (kunalfleethawks) [inline cypress/integration/address_customer.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_customer.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    cy.get('#addCustomerModal > .modal-dialog > .modal-content > .modal-footer > .btn-default').click();
+    cy.get('.address-book-icon').click();
+    cy.get('#customer-tab').click();
+    cy.wait(6000);
+});
+
+
+it('user should be able to edit ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/');
```
Comment (kunalfleethawks) [inline cypress/integration/address_customer.js]: Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_customer.js [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+
+it('validation error on required  fields ', function () {
+    /* ==== Generated with Cypress Studio ==== */
+    cy.visit('http://localhost:4200/');
+    cy.get(':nth-child(1) > .input-group > .form-control').clear();
+    cy.get(':nth-child(1) > .input-group > .form-control').type('rashmi.cypress123');
+    cy.get(':nth-child(2) > .input-group > .form-control').clear();
+    cy.get(':nth-child(2) > .input-group > .form-control').type('Hello#123');
+    cy.get('#btnsubmit').click();
```
Comment (kunalfleethawks) [inline cypress/integration/address_customer.js:136]: Take this url from cypress environment variable. Please refer to driver test cases

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
Take this url from cypress environment variable. Please refer to driver test cases

File: cypress/integration/address_customer.js (hunk 7 above)
Comment (rashmi0301) [reply cypress/integration/address_customer.js]: okay . I'll do it for all

File: cypress/integration/address_customer.js (hunk 9 above)
Comment (rashmi0301) [reply cypress/integration/address_customer.js:136]: Done

File: cypress/integration/address_customer.js (hunk 8 above)
Comment (rashmi0301) [reply cypress/integration/address_customer.js]: DONE

File: cypress/integration/address_broker.js (hunk 5 above)
Comment (rashmi0301) [reply cypress/integration/address_broker.js]: DONE

File: cypress/integration/address_broker.js (hunk 4 above)
Comment (rashmi0301) [reply cypress/integration/address_broker.js]: DONE

File: cypress/integration/address_broker.js (hunk 1 above)
Comment (rashmi0301) [reply cypress/integration/address_broker.js]: DONE

File: cypress/integration/address_broker.js (hunk 3 above)
Comment (rashmi0301) [reply cypress/integration/address_broker.js]: DONE

**Comment by rashmi0301** [comment]
@kunalfleethawks please review it  above changes and new test cases for user . 
Thanks

File: cypress/integration/address_broker.js (hunk 2 above)
Comment (rashmi0301) [reply cypress/integration/address_broker.js]: DONE

File: cypress/integration/address_customer.js (hunk 6 above)
Comment (rashmi0301) [reply cypress/integration/address_customer.js]: DONE

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**